python/ray/serve/backend_worker.py

```diff
@@ -251,6 +253,31 @@ def get_runner_method(self, request_item: Query) -> Callable:
             return self.callable
         return getattr(self.callable, method_name)
 
+    async def ensure_serializable_response(self, response: Any) -> Any:
+        if isinstance(response, starlette.responses.StreamingResponse):
+            # response contains a generator/iterator which is not serializable.
+            # Exhaust the generator/iterator and store the results in a buffer.
+            body_buffer = []
+
+            async def mock_send(message):
```

Comment:
can we check the the `message["type"]` can only be `http.response.body`? i don't want to silently miss a case.

Reply:
Yup, in the starlette source it only sends this specific type of message.  https://github.com/encode/starlette/blob/master/starlette/responses.py#L219